src/wallet/scriptpubkeyman.cpp

```diff
@@ -281,9 +281,10 @@ void LegacyScriptPubKeyMan::KeepDestination(int64_t index)
     KeepKey(index);
 }
 
-void LegacyScriptPubKeyMan::ReturnDestination(int64_t index, bool internal, const CPubKey& pubkey)
+void LegacyScriptPubKeyMan::ReturnDestination(int64_t index, bool internal, const CTxDestination& addr)
 {
-    ReturnKey(index, internal, pubkey);
+    ReturnKey(index, internal, m_reserved_key_to_index[index]);
+    m_reserved_key_to_index.erase(index);
```

Comment:
In commit "Key pool: Change ReturnDestination interface to take address instead of key" (06d411488c1c669f00a1c947416267941dbb7f7b)

I still don't understand why there is this distinction between ReturnKey and ReturnDestination. Why not just rename ReturnKey to ReturnDestination and get rid of the awkward wrapping?

Previous: https://github.com/bitcoin/bitcoin/pull/17304#discussion_r340311958

Reply:
The wrapper was for the original PR to make it clearer that things were being moved. As that no longer matters, I've added a commit to rename them.